resources/documentation/learn_ansible.rst

```diff
@@ -691,5 +691,395 @@ Now regarding to Jinja2:
+
+  - name: Get output from a shell command
+    shell: "cat /tmp/valkyrie"
+    register: my_result
+    ignore_errors: True
+
+  - name: Display content of the register
+    debug:
+      msg: "Profile"
+    loop: "{{my_result.stdout_lines}}"
+    when:
+      - item == "lenneth"
+      - sys_admin_is_master == "yes"  # <<< this is added as an AND with first condition
+
+Now execute playbook with an additional variable as extra vars, first with "no", then with "yes":
+
+.. code-block:: text
+
+  ansible-playbook ... -e sys_admin_is_master=no
+  ansible-playbook ... -e sys_admin_is_master=yes
+
+First time, you get:
+
+.. code-block:: text
+
+  TASK [debug : Display content of the register] **********************************************************************************************************************************************************************************************
+  skipping: [management1] => (item=freya)
+  skipping: [management1] => (item=lenneth)
+  skipping: [management1] => (item=odin)
+  skipping: [management1] => (item=loki)
+
+And second time:
+
+.. code-block:: text
+
+  TASK [debug : Display content of the register] **********************************************************************************************************************************************************************************************
+  skipping: [management1] => (item=freya)
+  ok: [management1] => (item=lenneth) =>
+    msg: Profile
+  skipping: [management1] => (item=odin)
+  skipping: [management1] => (item=loki)
+
+So it acted as an AND.
+
+Tags
+^^^^
+
+Tags act like the name: they tag part of the task, or role, so you can execute only this part, or skip it.
+
+For example:
+
+.. code-block:: yaml
+
+  - name: Display content of the register
+    debug:
+      msg: "Les sanglots longs"
+    tags:
+      - first
+
+  - name: Display content of the register
+    debug:
+      msg: "des violons de l'automne"
+    tags:
+      - second
+
+And at execution, use *--tags first,second* to execute both, *--tags second* to execute second only, or *--skip-tags first* that will skip the first one.
+
+Notify
+^^^^^^
+
+Sometime, you may wish that some actions take place at the end of each role, if a module returned a **changed** status.
+Best example: your role is dedicated to a service, and Ansible generate the configuration file using a template.
+You may want that this service is restarted if Ansible detect changes in the file when executing.
+
+Notify will be your friend for such cases.
+
+Create a new directory at root of your role, called **handlers**, and then inside a file called **main.yml**, with the following content:
+
+.. code-block:: yaml
+
+  - name: There was a change
+    debug:
+      msg: "THE FILE WAS CHANGED!! HOW DARE YOU!!"
+
+Note that the name of this task (There was a change) is important and will be targeted by the notify.
+
+Create also a template, called our_file.j2 in templates folder, with the following content:
```

Comment:
in **templates** folder (like **handlers** above)

Reply:
yes